wagtail/snippets/views/snippets.py

```diff
@@ -1151,9 +1177,26 @@ def get_urlpatterns(self):
 
         return urlpatterns + legacy_redirects
 
+    def register_admin_url_finder(self):
+        register_admin_url_finder(self.model, self.url_finder_class)
+
+    def register_chooser_widget(self):
+        # Set up admin model forms to use AdminSnippetChooser for any ForeignKey to this model
+        register_form_field_override(
+            ForeignKey,
+            to=self.model,
+            override={"widget": AdminSnippetChooser(model=self.model)},
+        )
+
+    def register_model_check(self):
+        def snippets_model_check(app_configs, **kwargs):
+            return check_panels_in_model(self.model, "snippets")
+
+        checks.register(snippets_model_check, "panels")
```

Comment:
Not sure if this should be here or left in `register_snippet()`...

Reply:
I think we can justify moving it here either way: the act of setting up CRUD views for the snippet is what makes the `panels` definition meaningful, and thus appropriate to have checks applied to it.

If anything, I think the code smell here is a symptom of the fact that the `panels` definition doesn't _reeeeally_ belong on the model. (We do already have a mechanism for defining `panels` through the ViewSet and having that take precedence over the one in the model, right? If not, we should - and as people get more comfortable with the idea of writing their own ViewSet definitions, I'd like to see that become standard practice. And, of course, if we do that, we should make sure that the checks here are being applied on the panels definition we're actually using.) Anyway, that's probably way out of scope for this PR :-)